### ISSUE TITLE
Error: Node inconsistent with parents

### ISSUE THREAD

**Issue by djhocking** [djhocking/Bird_Communities#2]
The `Scripts/Abundance_Kery_Royle.R` is a stand alone script based on a script from Marc Kery and Andy Royle's Hierarchical Modeling Book (Volume 1, 9.3), but it does not work with this data currently. Error: 

```
Error in checkForRemoteErrors(val) : 
  3 nodes produced errors; first error: Error in node dclass[17]
Node inconsistent with parents
```

Potentially the starting values don't work with the data-model combination. The exact node changes with the random starting values. Another potential could be if the order of `dclass` and `tint` somehow don't match up with the order of the other data. I'm quite stuck on what to try next.

Sometimes I get a slightly different error:

```
Error in checkForRemoteErrors(val) : 
  3 nodes produced errors; first error: Error in node pi.pd.c[1,158]
Invalid parent values
```

**Comment by djhocking** [comment]
My experience with this error in the past is that there is an inappropriate/unacceptable value that leads to the calculation of the node being mentioned in the error.  Sometimes this can be something like a negative value being generated for a parameter that needs to be positive (like a variance parameter), a value outside a 0-1 range being generated for a probability, or a negative value being supplied for something log-scale.  It can be difficult to trace the problem all the way back to the source because it can occur at any step that leads to the node mentioned in the error, even in the starting values.  Sometimes you can try troubleshooting by fixing parts of the calculation (like “pdet”) to a reasonable value, commenting out the steps to generate it, and seeing if the problem persists in the other part of the calculation (pi.pd).  I tried that some without success but that’s the only way I know to deal with these errors without really carefully thinking about all the data objects, model calculations, and starting values in turn.

**Comment by djhocking** [comment]
**Options:**

1. Add overdispersion terms to see if that gives sufficient flexibility for fitting
2. See if there's anywhere to restrict priors to prevent values outside the ranges for a probability
3. Start with a simple model for just 2016 visit 2 and see if it works. It could be that the observer got better at removal sampling over time and controlling distance sampling, so the early depletion curves were insufficient.
4. Double check (again) the data form to ensure that the observation and point level data match
5. Test with simulation data and alter simulations to see when it breaks
6. Analyze point count with time depletion (removal) but without detection decaying with distance. I could do this for the whole point (up to 150 m) or restrict it to just 50 or 100 m where detection under these conditions were unlikely to vary greatly.

These are not mutually exclusive at all. I might try number 6 first since it's the easiest to do and most likely to work. Number 3 would be my next step. I've tried some of the others to a moderate extent.

**Comment by djhocking** [comment]
I decided to do a little more investigation (option 2 above) related to distance sampling since that was where the problems occurred. I found that if I restricted `sigma[s]` to 0-14.9 it worked but at 15 it failed. I looked at some more of the data generation from the new Kery and Royle Hierarchical Modeling book and found that distance was often scaled (divided by 100) in distance sampling. This had the same results and I was able to add the covariates back onto `log(sigma[s])` and it worked and the answers all made sense. Now to just finalize the covariates in the model then loop through the species and organize the output.
